extension/core_functions/scalar/list/list_reduce.cpp

```diff
@@ -184,7 +184,7 @@ static unique_ptr<FunctionData> ListReduceBind(ClientContext &context, ScalarFun
 
 	// the list column and the bound lambda expression
 	D_ASSERT(arguments.size() == 2);
-	if (arguments[1]->expression_class != ExpressionClass::BOUND_LAMBDA) {
+	if (arguments[1]->GetExpressionClass() != ExpressionClass::BOUND_LAMBDA) {
```

Comment:
We could also consider adding a `IsInstanceOf(ExpressionClass& class)` shorthand here